### PR TITLE
databricks: migrate metrics setup to metadata.yaml

### PR DESCRIPTION
**Pull request by pmcollins**

### CODE REVIEW

File: internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+			metric.SetUnit("ms")
+			metric.SetDataType(pdata.MetricDataTypeGauge)
+		},
+	},
+	&metricImpl{
+		"databricks.jobs.schedule.status",
+		func(metric pdata.Metric) {
+			metric.SetName("databricks.jobs.schedule.status")
+			metric.SetDescription("A snapshot of the pause/run status per job taken at each scrape")
+			metric.SetUnit("{status}")
```
Comment (rmfitzpatrick) [inline internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go:103]: same `{jobs}` suggestion/question

File: internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+// Metrics contains a set of methods for each metric that help with
+// manipulating those metrics.
+var Metrics = &metricStruct{
+	&metricImpl{
+		"databricks.jobs.active.total",
+		func(metric pdata.Metric) {
+			metric.SetName("databricks.jobs.active.total")
+			metric.SetDescription("A snapshot of the number of active jobs taken at each scrape")
+			metric.SetUnit("{tot}")
```
[the remark this answers is not in the document]
Comment (pmcollins) [reply internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go:85]: sure -- will fix

**Comment by rmfitzpatrick** [comment]
Maybe a taller ask for another time, but is it possible to add the generation commands to the makefile for easy updates?

**Comment by pmcollins** [comment]
> Maybe a taller ask for another time, but is it possible to add the generation commands to the makefile for easy updates?

Seems like a fine idea but perhaps non-trivial? esp since we don't have makefiles per receiver (should we?). Perhaps we should punt on this for this PR (would like to get it merged before the release).

File: internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go (hunk 1 above)
Comment (pmcollins) [reply internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go:103]: There was another {tot} for databricks.jobs.total -- probably what you were referring to, sorry. Changed to {jobs}.

File: internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pmcollins) [reply internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go]: fixed

File: internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go (hunk 1 above)
Comment (rmfitzpatrick) [reply internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go:103]: My mistake, I thought it was jobs per a given status.

**Comment by rmfitzpatrick** [comment]
> Seems like a fine idea but perhaps non-trivial? esp since we don't have makefiles per receiver (should we?). Perhaps we should punt on this for this PR (would like to get it merged before the release).

First impression was that it could be done with a list of applicable receivers in the main makefile and not require adding one per component. Makes sense to wait for another adopter or so.

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**